web/src/presence.ts

```diff
@@ -304,6 +324,58 @@ export function last_active_date(user_id: number): Date | undefined {
     return new Date(info.last_active * 1000);
 }
 
+export function map_status(status: string): "active" | "idle" | "offline" {
+    switch (status) {
+        case "active":
+        case "idle":
+        case "offline":
+            return status;
+        default:
+            return "offline";
+    }
+}
+
+export let fetch_user_last_seen_status = (user_id: number): void => {
+    const url = `json/users/${user_id}/presence`;
+
+    channel.get({
+        url,
+        success(data: unknown) {
+            const parsed_data = user_last_seen_response_schema.safeParse(data);
+
+            if (parsed_data.success) {
+                const response = parsed_data.data;
+
+                if (response.result === "success" && response.presence) {
+                    const {aggregated} = response.presence;
+                    presence_info.set(user_id, {
+                        status: map_status(aggregated.status),
+                        last_active: aggregated.timestamp,
+                    });
+
+                    // Update the user's last seen time in the popover once we have the data
+                    $(".user-last-seen-time").text(
+                        timerender.last_seen_status_from_date(
+                            new Date(aggregated.timestamp * 1000),
+                        ),
+                    );
+                } else {
+                    // fallback logic for users who haven't generated any presence data
+                    const user = people.get_by_user_id(user_id);
+                    presence_info.set(user_id, {
+                        status: "offline",
+                        last_active: new Date(user.date_joined).getTime() / 1000,
```

Comment:
If the request fails for some reason like there is no `UserPresence` object for a user, then we would need to change the "Loading" text to something, but am not sure.

Reply:
We could display the error message returned from server, but we have limited space. So, would be good to try it out once.